src/protagonist/API/Features/DeliveryChannels/DeliveryChannelPolicyRepository.cs

```diff
@@ -0,0 +1,35 @@
+﻿using DLCS.Core.Caching;
+using DLCS.Model.DeliveryChannels;
+using DLCS.Model.Policies;
+using DLCS.Repository;
+using LazyCache;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace API.Features.DeliveryChannels;
+
+public class DeliveryChannelPolicyRepository : IDeliveryChannelPolicyRepository
+{
+    private readonly DlcsContext dlcsContext;
+
+    public DeliveryChannelPolicyRepository(
+        DlcsContext dlcsContext)
+    {
+        this.dlcsContext = dlcsContext;
+    }
+
+    public DeliveryChannelPolicy RetrieveDeliveryChannelPolicy(int customer, string channel, string policy)
```

Comment:
As discussed - make less DB trips, this request could be fulfilled with a single DB trip. A better way may be to have customer-level caching of _all the policies this customer could possibly have_ and then do in-memory filtering to get the final result.